lib/Serialization/ModuleFile.cpp

```diff
@@ -555,6 +555,8 @@ bool ModuleFile::readIndexBlock(llvm::BitstreamCursor &cursor) {
       break;
     }
   }
+
+  llvm_unreachable("Handled BitstreamEntry::EndBlock in the above switch.");
```

Comment:
I've been putting off dealing with these, but really I think I'd prefer to restructure the loop (here and below) so that it actually makes sense, either by using `while (true)` or by handling a malformed file.